Core/Pixel.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Pixel.swift
+//  DuckDuckGo
+//
+//  Copyright © 2018 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Alamofire
+
+public enum PixelDefinition: String {
+    
+    case appLaunch = "ml"
+    case forgetAllExecuted = "mf"
+    case privacyDashboardOpened = "mp"
+    
+}
+
+public protocol PixelFiring {
+    
+    func fire(pixel: PixelDefinition)
+    
+}
+
+public class Pixel: PixelFiring {
+
+    private static let pixel = Pixel()
+    
+    private let appUrls = AppUrls()
+    
+    public static func shared() -> Pixel {
```

Comment:
Do you need this? Why not expose the property directly?

Reply:
Was future proofing as I didn't know if I'd have to do batching. That's not happening now - these functions could just be static even?

Reply:
I'm not clear on how that helps future proof. If you're keeping the shared property, I would like to understand. But it sounds like you're removing it, in which case, it doesn't matter.

Reply:
I guess using `shared()` is old-school now. It's a standard pattern and hides implementation details so that you can defer architectural changes until later, is all. I am just pondering what the implications of using the static var directly are if any - guess you can keep the same API and still change the implementation using a dynamic getter.  Anyway yeah, I think I'm going to ditch it.

Reply:
That's exactly where my thinking was too - because we're in Swift land we don't need to wrap things in methods to protect against future changes.